dataprep/clean/clean_date.py

```diff
@@ -0,0 +1,996 @@
+    df = df.apply(
+        format_date,
+        args=(
+            col,
+            target_format,
+            {"origin_tz": origin_timezone, "target_tz": target_timezone},
+            fix_empty,
+        ),
+        axis=1,
+        meta=meta,
+    )
+    df, nrows = dask.compute(df, df.shape[0])
+    # output the report describing the changes to the column
+    if show_report:
+        create_report("Date", STATS, nrows)
+    return df
+
+
+def format_date(
+    row: pd.Series,
+    col: str,
+    target_format: str,
+    tz_info: Dict[str, Union[str, int]],
+    fix_empty: str,
+) -> pd.Series:
+    """
+    This function cleans date string.
+    Parameters
+    ----------
+    df, col, target_format, tz_info, fix_empty
+        same as explained in clean_date function
+    """
+    date = row[col]
+    origin_timezone = tz_info["origin_tz"]
+    target_timezone = tz_info["target_tz"]
+    if check_date(date) == "null":
+        STATS["null"] += 1
+        row[f"{col}_clean"] = np.nan
+    elif check_date(date) == "unknown":
+        STATS["unknown"] += 1
+        row[f"{col}_clean"] = np.nan
+    elif check_date(date) == "cleaned":
+        # Handle date data and timezone
+        parsed_date_data = parse(date=date, fix_empty=fix_empty)
+        parsed_date_data.set_tzinfo(timezone=origin_timezone)
+        parsed_date_data = set_timezone_offset(origin_timezone, parsed_date_data)
+        # Handle target format and timezone
+        parsed_target_format_data = check_target_format(target_format=target_format)
+        parsed_target_format_data = set_timezone_offset(target_timezone, parsed_target_format_data)
+        if parsed_target_format_data.valid:
+            if parsed_date_data.valid == "cleaned":
+                transformed_date = transform(
+                    parsed_date_data=parsed_date_data,
+                    parsed_target_format_data=parsed_target_format_data,
+                    target_format=target_format,
+                    target_timezone=target_timezone,
+                )
+                row[f"{col}_clean"] = f"{transformed_date}"
+                if row[col] != row[f"{col}_clean"]:
+                    STATS["cleaned"] += 1
+            else:
+                STATS["unknown"] += 1
+                row[f"{col}_clean"] = np.nan
+        else:
+            raise ValueError(
+                f"target_format {target_format} is invalid. "
+                f"Invalid tokens are {parsed_target_format_data.invalid_tokens}. "
+                f"Please retype it."
+            )
+    return row
+
+
+def set_timezone_offset(
+    timezone: Union[str, Any], parsed_data: Union[ParsedDate, ParsedTargetFormat, Any]
+) -> Any:
+    """
+    This function set timezone information for parsed date or parsed target format
+    Parameters
+    ----------
+    timezone
+        string name of timezone
+    parsed_data
+        parsed date or parsed target format
+    """
+    example_date = datetime.datetime(2009, 9, 1)
```

Comment:
Can you explain the objective of setting `example_date` here

Reply:
Because we need to draw the offset hours and minutes comparing with UTC using pytz:
```python
pytz_offset = pytz.timezone(timezone).utcoffset(example_date)
```
And the function utcoffset need a datetime parameter. The value of the paramter is not important. However, we must need this datetime value to get the result we want.